source/common/http/utility.h

```diff
@@ -182,6 +182,14 @@ void appendXff(RequestHeaderMap& headers, const Network::Address::Instance& remo
  */
 void appendVia(RequestOrResponseHeaderMap& headers, const std::string& via);
 
+/**
+ * Update authority with the specified hostname and copy the original authority to the forwarded
+ * host header.
+ * @param headers headers where authority should be updated.
+ * @param hostname hostname that authority should be updated with.
+ */
+void updateAuthority(RequestHeaderMap& headers, absl::string_view hostname);
```

Comment:
have you checked all call sites for setHost?  I see at least one other in extensions which looks like it might need to be updated.

Reply:
I think this comes down to whether we want to make just part of router host rewrite determined by router config or a global behavior that applies to filter that rewrite host for some reason. If we make this behavior dependent on the router config then extensions need to either add their own config options that control XFH or be modified to always do it. If we go with this change I would rather make it in separate PRs and scope this one just to the router filter.

I think I have a preference for scoping this PR to the router hots rewrite only and not extend it to the filters.